### DISCUSSION TITLE
How to get head URL in github action

### DISCUSSION THREAD

**Discussion by molly-firebolt** [community/community#29837] [Actions] [answered]
I'm trying to write an action that creates a new branch in a staging repo and pulls in changes when a PR is opened against our main repo. 

How do I dynamically reference the head URL? I've tried github.head_url and github.event.pull_request.head.url with no luck, to put in this command: git pull ${{ github.event.pull_request.head.url }} ${{ github.head_ref }}

head_ref works great, I just can't figure out how to reference the forked repo url.

**Comment by airtower-luna** [comment, marked as answer]
The code for what you already have got a bit messed up, evidently the markdown parser assumed you have some TeX formula in there. Guessing from what remains and looking at the [`pull_request` payload example](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-34) you're pretty close.

`github.event.pull_request.head.repo.clone_url` is probably what you want, or `ssh_url` at the end if you want to authenticate using a public key instead of a token.

**Reply by molly-firebolt** [reply]
Exactly it, thank you so much!
